substrate/frame/examples/single-block-migrations/src/migrations/v1.rs

```diff
@@ -0,0 +1,223 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use frame_support::{
+	storage_alias,
+	traits::{Get, OnRuntimeUpgrade},
+};
+
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+/// Collection of storage item formats from the previous storage version.
+///
+/// Required so we can read values in the old storage format during the migration.
+mod old {
+	use super::*;
+
+	/// V0 type for [`crate::Value`].
+	#[storage_alias]
+	pub type Value<T: crate::Config> = StorageValue<crate::Pallet<T>, u32>;
+}
+
+/// Private module containing *version unchecked* migration logic.
+///
+/// Should only be used by the [`VersionedMigration`](frame_support::migrations::VersionedMigration)
+/// type in this module to create something to export.
+///
+/// The unversioned migration should be kept private so the unversioned migration cannot
+/// accidentally be used in any runtimes.
+///
+/// For more about this pattern of keeping items private, see
+/// - <https://github.com/rust-lang/rust/issues/30905>
+/// - <https://internals.rust-lang.org/t/lang-team-minutes-private-in-public-rules/4504/40>
+mod version_unchecked {
+	use super::*;
+
+	/// Implements [`OnRuntimeUpgrade`], migrating the state of this pallet from V0 to V1.
+	///
+	/// In V0 of the template [`crate::Value`] is just a `u32`. In V1, it has been upgraded to
+	/// contain the struct [`crate::CurrentAndPreviousValue`].
+	///
+	/// In this migration, update the on-chain storage for the pallet to reflect the new storage
+	/// layout.
+
+	pub struct MigrateV0ToV1<T: crate::Config>(sp_std::marker::PhantomData<T>);
+
+	impl<T: crate::Config> OnRuntimeUpgrade for MigrateV0ToV1<T> {
+		/// Return the existing [`crate::Value`] so we can check that it was correctly set in
+		/// `version_unchecked::MigrateV0ToV1::post_upgrade`.
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			use codec::Encode;
+
+			// Access the old value using the `storage_alias` type
+			let old_value = old::Value::<T>::get();
+			// Return it as an encoded `Vec<u8>`
+			Ok(old_value.encode())
+		}
+
+		/// Migrate the storage from V0 to V1.
+		///
+		/// - If the value doesn't exist, there is nothing to do.
+		/// - If the value exists, it is read and then written back to storage inside a
+		/// [`crate::CurrentAndPreviousValue`].
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			// Read the old value from storage
+			if let Some(old_value) = old::Value::<T>::get() {
```

Comment:
I think in general it should use `take` here, just to ensure that there is no old storage remaining.  
As there could be edge-cases where a `None` in the storage is treated as "nothing is in storage".